pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -154,12 +157,31 @@ def restore_model(self) -> None:
 
     def restore_model_weights(self, checkpoint_path: Optional[Union[str, Path]]) -> None:
         """ Restore only the model weights. """
+        log.info("Restoring model weights from checkpoint")
         checkpoint = self._loaded_checkpoint
+        if hasattr(self.trainer.training_type_plugin, "num_processes"):
```

Comment:
I would prefer this logic to life directly inside the FSDP plugin. This would definitely be cleaner.